lisa/tools/cargo.py

```diff
@@ -110,23 +156,44 @@ def test(
         cwd: Optional[PurePath] = None,
     ) -> ExecutableResult:
         echo = self.node.tools[Echo]
-        original_path = echo.run(
+        path = echo.run(
             "$PATH",
             shell=True,
             expected_exit_code=0,
             expected_exit_code_failure_message="failure to grab $PATH via echo",
         ).stdout
-        home_path = echo.run(
-            "$HOME",
-            shell=True,
-            expected_exit_code=0,
-            expected_exit_code_failure_message="failure to grab $HOME path",
-        ).stdout
-        new_path = f"{home_path}/.cargo/bin:{original_path}"
+
+        if os.path.dirname(self.command) not in path:
+            path = f"{os.path.dirname(self.command)}:{path}"
```

Comment:
Why do we need this? `self.command` contains the absolute path to the binary right?

Reply:
@anirudhrb  self.command will have absolute path but there are rustc binary that need to be there in PATH of cargo we are using for given toolchain otherwise it will throw error something like **not able to detect rustc version**. You can check for different pipeline run instance for that error.